src/de.hpi.swa.graal.squeak/src/de/hpi/swa/graal/squeak/model/ContextObject.java

```diff
@@ -324,7 +325,11 @@ public AbstractSqueakObject getSender() {
         final Object value = FrameAccess.getSender(truffleFrame);
         if (value instanceof FrameMarker) {
             getBlockOrMethod().getDoesNotNeedSenderAssumption().invalidate("Sender requested");
-            return ((FrameMarker) value).getMaterializedContext();
+            try {
+                return ((FrameMarker) value).getMaterializedContext();
+            } catch (final SqueakException e) {
+                return NilObject.SINGLETON;
+            }
```

Comment:
I don't think it's a good idea to catch `SqueakException` here, it's probably better to move this logic into `getMaterializedContext()`.